src/fsverity/ioctl.rs

```diff
@@ -71,8 +75,33 @@ pub fn fs_ioc_measure_verity<F: AsFd, H: FsVerityHashValue>(fd: F) -> Result<H>
     }
 
     if digest.digest_algorithm != digest_algorithm || digest.digest_size != digest_size {
-        Err(std::io::Error::from(std::io::ErrorKind::InvalidData))?
+        Err(Errno::ILSEQ)
     } else {
         Ok(digest.digest)
     }
 }
+
+/// Like [`fs_ioc_measure_verity`] but does not fail if the file does not have fsverity enabled.
+pub fn fs_ioc_measure_verity_optional<F: AsFd, H: FsVerityHashValue>(fd: F) -> Result<Option<H>> {
```

Comment:
...or maybe just return `Option<H>>` from the other version of the call, and update the users of that API?  That would let us punt on the errno/anyhow/thiserror thing.